commitizen/cz/conventional_commits/conventional_commits.py

```diff
@@ -190,8 +190,11 @@ def schema(self) -> str:
 
     def schema_pattern(self) -> str:
         PATTERN = (
-            r"(build|ci|docs|feat|fix|perf|refactor|style|test|chore|revert|bump)"
-            r"(\(\S+\))?!?:(\s.*)"
+            r"(?s)"  # To explictly make . match new line
```

Comment:
I'm not familiar with re enough to understand what this means. Could you please describe a bit more on why this is added?

Reply:
rest of the stuff looks good to me  👍

Reply:
`(?s)` is an inline flag, which is the same as `re.DOTALL` ([ref](https://docs.python.org/3/library/re.html#re.DOTALL)). This is needed because I add a `$` to make this schema pattern to match the whole commit message instead of matching the first line only. If the flag is not set, `\n\n.*` only allows one line of message body.

You can use the following code to verify that behavior:
```python
import re

PATTERN = (
    r"(?s)"
    r"(build|ci|docs|feat|fix|perf|refactor|style|test|chore|revert|bump)"  # type
    r"(\(\S+\))?!?:"  # scope
    r"( [^\n\r]+)"  # subject
    r"((\n\n.*)|(\s*))?$"
)

# Shouldn't be `None`
print(
    re.match(
        PATTERN,
        (
            "refactor(docs): remove README.rst and use docs/README.md\n\n"
            "By removing README.rst, we no longer need to maintain "
            "two document with almost the same content\n"
            "Github can read docs/README.md as README for the project."
        ),
    ),
)
# Should be None
print(
    re.match(PATTERN, "feat: an awesome feature\nwith an ill-formated commit message"),
)
```

Reply:
@Lee-W Should I add more explanation to that regex pattern?

Reply:
nope, but I think we could add a link as comment to this issue.

---

Thanks for the explanation! Now I understand it. Just want to make sure, did you forget to remove the "(?s)" in this example? If I remove "($s)", then the result is just like what you describe.